_posts/2022-08-15-Jeffy-Jeffy.md

```diff
@@ -0,0 +1,29 @@
+
+###### Statiscally typed: 
+C++ is a statistically typed language, meaning that the variable types are known at the time of compile. One of the biggest advantages of doing so is that all types of "screening" can be conducted by the compiler, which allows for a lot of trivial bugs to be caught at an early stage, that is without running the progrm. Thus, it may improve runtime efficiency. I think one of the disadvantages of this is that we can no longer change the type of the variable as the code is being utilized. Speaking of variables, some of the allowed variables in C++ are bool, int, double (similar to float), std::string, std::vector, and std::map. Overall, I believe that the true benefit of statistically typed language is more pronounced with longer and more complex programs. 
+
+###### Compiled programming: 
+Compiling refers to the transformation from human readable code into a machine executable code. Often when we refer to compiling, what we truly mean is both compiling and linking, where linking combines all the compiled data into a sinle library or executable. C++ is a compiled language; however, python is an interpreted language. This means that when using python, the code is first interpreted, then compiled -- adding an additional step. This makes C++ much more faster and efficient way of programming.  Additionally, compiling also lets us do optimizations which are not possible in interpretted languages like Python. This means removing unused codes, reordering operations, and performing simple mathematical operations. Again, I think this is also one of those functions that become much more helpful when writing complex programs. 
+
+###### Basics of C++ syntax: 
+One of the most important differences in syntax between python and C++ is that C++ ends each statement with a semi colon. I think some of the other things that we will have to get used to is commenting using "//". Lastly, functions are very different, in that both the type of the return values as well as the input parameters must be mentioned as functions are declared. Also, can't forget curly braces to separate different chunks of code! 
+
+
+###### Discussion Questions: 
+1. I think C++ is much more helpful with numerically heavy problems as performance is always a concern with such programs and C++ can help with that. Additionally, programs that are long and complex can be completed using C++ easily as it is a compiled languages and thus, bugs and trivial mistakes are easily identified so we don't have to spend too much time debugging. I believe Python is much more helpful with smaller projects and scripts. File input and output is also much more easier with Python. 
```

Comment:
Add on to that, the existence of packages such as NumPy, Pandas, SciKit Learn on Python, is hardly found on any other languages. The prevalence and dominance of those packages are sometimes good reasons to use Python by themselves.